lib/std/linked_list.zig

```diff
@@ -155,6 +155,88 @@ test "basic SinglyLinkedList test" {
     testing.expect(list.first.?.next.?.next == null);
 }
 
+/// Simpler singly-linked list with no seperate head structure.
+/// Instead the last node points to to the first. 
+///
+/// Using a head without surrounding struct is optional,
+/// otherwise the first element can be used as the head.
+///
+/// Use with @fieldParentPtr(), like so:
+/// 
+/// const Number = struct {
+///     list: ListHead,
+///     value: i32,
+/// };
+/// fn number(list: *ListHead) Number {
+///     return @fieldParentPtr(Number, "list", list);
+/// }
+pub const ListHead = struct {
+    next: *ListHead,
+
+    pub fn init(list: *ListHead) void {
+        list.next = list;
+    }
+
+    pub fn insert(list: *ListHead, new: *ListHead) void {
+        new.next = list.next;
+        list.next = new;
+    }
+
+    pub fn last(list: *ListHead) *ListHead {
+        var cur = list;
+        while (cur.next != list) : (cur = cur.next) {}
+        return cur;
+    }
+
+    /// Insert at end
+    pub fn push(list: *ListHead, new: *ListHead) void {
+        list.last().insert(new);
+    }
+
+    /// Read from end, and remove from list
+    pub fn pop(list: *ListHead) *ListHead {
+        var cur = list;
+        var prev = list;
+        while (cur.next != list) : (cur = cur.next) {
+            prev = cur;
+        }
+        prev.next = list;
+        return cur;
+    }
+
+    pub fn isEmpty(list: *ListHead) bool {
+        if (list.next == null) return true;
```

Comment:
`.next` isn't optional; how could this work?

Reply:
oh whoops